cpp/oneapi/dal/backend/primitives/test/ndarray.cpp

```diff
@@ -416,6 +416,23 @@ TEST("can cast ndarray to ndview", "[ndarray]") {
     REQUIRE(x_view.get_strides() == x.get_strides());
 }
 
+TEST_M(ndarray_test, "can be flattened host", "[ndarray]") {
+    const std::int64_t m = 9;
+    const std::int64_t n = 5;
+
+    std::vector<float> data_vector(m * n, 1.0f);
+
+    auto x = ndarray<float, 2>::wrap(data_vector.data(), { m, n });
+
+    auto raw_arr = x.flatten();
+
+    REQUIRE(raw_arr.get_count() == (m * n));
+
+    for (std::int64_t i = 0; i < (m * n); ++i) {
+        REQUIRE(raw_arr[i] == 1.0f);
+    }
+}
```

Comment:
Comment regarding separation on code blocks in unit tests. Now it's difficult to follow test flow. Here is some good practice.
https://softwareengineering.stackexchange.com/a/15934

```c++
TEST("...") {
    // Some lines 
    // of setup code
    // go here.

    // The action being tested happens after a blank line.

    // An assertion follows another blank line.
}
```

This is applicable to all unit tests.